From 4ee6795432fc9f20a7d64ec185660290f109676d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bastien=20ROUCARI=C3=88S?= <roucaries.bastien@gmail.com>
Date: Sat, 31 Dec 2016 15:09:43 +0100
Subject: [PATCH] Add a new option in order to use desktop file

Forwarded: yes, https://github.com/ImageMagick/ImageMagick/issues/331
bug: https://github.com/ImageMagick/ImageMagick/issues/331
bug-debian: https://bugs.debian.org/773426
---
 wand/display.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/wand/display.c b/wand/display.c
index b1a112e70..c9248ab49 100644
--- a/wand/display.c
+++ b/wand/display.c
@@ -184,6 +184,7 @@ static MagickBooleanType DisplayUsage(void)
       "-label string        assign a label to an image",
       "-limit type value    pixel cache resource limit",
       "-loop iterations     loop images then exit",
+      "-nostdin             do not try to open stdin",
       "-map type            display image using this Standard Colormap",
       "-matte               store matte channel if the image has one",
       "-monitor             monitor progress",
@@ -314,6 +315,7 @@ WandExport MagickBooleanType DisplayImageCommand(ImageInfo *image_info,
     image_stack[MaxImageStackDepth+1];
 
   MagickBooleanType
+    nostdin,
     fire,
     pend,
     respect_parenthesis;
@@ -377,6 +379,7 @@ WandExport MagickBooleanType DisplayImageCommand(ImageInfo *image_info,
   option=(char *) NULL;
   pend=MagickFalse;
   respect_parenthesis=MagickFalse;
+  nostdin=MagickFalse;
   resource_database=(XrmDatabase) NULL;
   (void) ResetMagickMemory(&resource_info,0,sizeof(resource_info));
   server_name=(char *) NULL;
@@ -413,6 +416,9 @@ WandExport MagickBooleanType DisplayImageCommand(ImageInfo *image_info,
           ThrowDisplayException(OptionError,"MissingArgument",option);
         server_name=argv[i];
       }
+    if (LocaleCompare("nostdin",option+1) == 0) {
+      nostdin = MagickTrue;
+    }
     if ((LocaleCompare("help",option+1) == 0) ||
         (LocaleCompare("-help",option+1) == 0))
       return(DisplayUsage());
@@ -460,7 +466,7 @@ WandExport MagickBooleanType DisplayImageCommand(ImageInfo *image_info,
       if (image != (Image *) NULL)
         break;
       else
-        if (isatty(STDIN_FILENO) != MagickFalse)
+        if (isatty(STDIN_FILENO) != MagickFalse || nostdin == MagickTrue)
           option="logo:";
         else
           option="-";
@@ -1425,6 +1431,8 @@ WandExport MagickBooleanType DisplayImageCommand(ImageInfo *image_info,
           break;
         if (LocaleCompare("normalize",option+1) == 0)
           break;
+	if (LocaleCompare("nostdin",option+1) == 0)
+	  break;
         ThrowDisplayException(OptionError,"UnrecognizedOption",option);
       }
       case 'p':
