From 5eae1b27ddc10e6f158b4a4990b598a131488c95 Mon Sep 17 00:00:00 2001
From: Dirk Lemstra <dirk@git.imagemagick.org>
Date: Mon, 19 Jun 2017 23:00:54 +0200
Subject: [PATCH] Fixed potential memory leak in screenshot coders

bug-debian: https://bugs.debian.org/867897
bug: https://github.com/ImageMagick/ImageMagick/issues/556
origin: https://github.com/ImageMagick/ImageMagick/commit/8c10b9247509c0484b55330458846115131ec2ae
---
 coders/screenshot.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/coders/screenshot.c b/coders/screenshot.c
index aaa3bbde9..7cf40c4fc 100644
--- a/coders/screenshot.c
+++ b/coders/screenshot.c
@@ -165,16 +165,16 @@ static Image *ReadSCREENSHOTImage(const ImageInfo *image_info,
       screen->columns=(size_t) GetDeviceCaps(hDC,HORZRES);
       screen->rows=(size_t) GetDeviceCaps(hDC,VERTRES);
       screen->storage_class=DirectClass;
+      if (image == (Image *) NULL)
+        image=screen;
+      else
+        AppendImageToList(&image,screen);
       status=SetImageExtent(screen,screen->columns,screen->rows);
       if (status == MagickFalse)
         {
           InheritException(exception,&image->exception);
           return(DestroyImageList(image));
         }
-      if (image == (Image *) NULL)
-        image=screen;
-      else
-        AppendImageToList(&image,screen);
 
       bitmapDC=CreateCompatibleDC(hDC);
       if (bitmapDC == (HDC) NULL)
