From 6acc9880b745bbf4655314715455f089c520af0a Mon Sep 17 00:00:00 2001
From: Glenn Randers-Pehrson <glennrp@gmail.com>
Date: Sun, 23 Jul 2017 11:53:53 -0400
Subject: [PATCH] coders/png.c: Memory leak Fixed Issue 600

bug-debian: https://bugs.debian.org/870116
bug: https://github.com/ImageMagick/ImageMagick/issues/600
origin:  https://github.com/ImageMagick/ImageMagick/commit/19b03f36c4807bc2b8a3a51718ce026fc700eb54
---
 coders/png.c | 15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/coders/png.c b/coders/png.c
index 082bdb849..b6a0cba45 100644
--- a/coders/png.c
+++ b/coders/png.c
@@ -1970,6 +1970,9 @@ static Image *ReadOnePNGImage(MngInfo *mng_info,
     length,
     row_offset;
 
+  Quantum
+    *volatile quantum_scanline;
+
   ssize_t
     j;
 
@@ -2131,13 +2134,16 @@ static Image *ReadOnePNGImage(MngInfo *mng_info,
       */
       png_destroy_read_struct(&ping,&ping_info,&end_info);
 
+      if (pixel_info != (MemoryInfo *) NULL)
+        pixel_info=RelinquishVirtualMemory(pixel_info);
+
+      if (quantum_scanline != (Quantum *) NULL)
+        quantum_scanline=(Quantum *) RelinquishMagickMemory(quantum_scanline);
+
 #ifdef IMPNG_SETJMP_NOT_THREAD_SAFE
       UnlockSemaphoreInfo(ping_semaphore);
 #endif
 
-      if (pixel_info != (MemoryInfo *) NULL)
-        pixel_info=RelinquishVirtualMemory(pixel_info);
-
       if (logging != MagickFalse)
         (void) LogMagickEvent(CoderEvent,GetMagickModule(),
           "  exit ReadOnePNGImage() with error.");
@@ -3232,9 +3238,6 @@ static Image *ReadOnePNGImage(MngInfo *mng_info,
 
     for (pass=0; pass < num_passes; pass++)
     {
-      Quantum
-        *quantum_scanline;
-
       register Quantum
         *r;
 
