From b251fb91d97524de94e3abe110a4735a847a95a6 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sun, 16 Jul 2017 20:34:52 -0400
Subject: [PATCH] memory leak in WriteCALSImage

bug-debian: https://bugs.debian.org/870475
bug: https://github.com/ImageMagick/ImageMagick/issues/571
origin: https://github.com/ImageMagick/ImageMagick/commit/3d5ac8c20846871f1eb3068b65f93da7cd33bfd0
---
 coders/cals.c | 2 ++
 coders/jp2.c  | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/coders/cals.c b/coders/cals.c
index 2be09e54b..a83cca912 100644
--- a/coders/cals.c
+++ b/coders/cals.c
@@ -564,6 +564,7 @@ static MagickBooleanType WriteCALSImage(const ImageInfo *image_info,
   group4_image=CloneImage(image,0,0,MagickTrue,&image->exception);
   if (group4_image == (Image *) NULL)
     {
+      write_info=DestroyImageInfo(write_info);
       (void) CloseBlob(image);
       return(MagickFalse);
     }
@@ -572,6 +573,7 @@ static MagickBooleanType WriteCALSImage(const ImageInfo *image_info,
   group4_image=DestroyImage(group4_image);
   if (group4 == (unsigned char *) NULL)
     {
+      write_info=DestroyImageInfo(write_info);
       (void) CloseBlob(image);
       return(MagickFalse);
     }
diff --git a/coders/jp2.c b/coders/jp2.c
index 7503302bf..60365b869 100644
--- a/coders/jp2.c
+++ b/coders/jp2.c
@@ -924,7 +924,7 @@ static MagickBooleanType WriteJP2Image(const ImageInfo *image_info,Image *image)
       &parameters.subsampling_dx,&parameters.subsampling_dy);
   property=GetImageProperty(image,"comment");
   if (property != (const char *) NULL)
-    parameters.cp_comment=(char *)property;
+    parameters.cp_comment=(char *) property;
   channels=3;
   jp2_colorspace=OPJ_CLRSPC_SRGB;
   if (image->colorspace == YUVColorspace)
