Description: Pass correct path to dlopen
Author: Sebastian Ramacher <sramacher@debian.org>
Forwarded: not-needed
Last-Update: 2015-05-21

--- a/source/encoder/api.cpp
+++ b/source/encoder/api.cpp
@@ -288,7 +288,7 @@
 #define ext ".dylib"
 #else
 #include <dlfcn.h>
-#define ext ".so"
+#define ext ".so." xstr(X265_BUILD)
 #endif
 
 extern "C"
@@ -302,9 +302,9 @@
         if (bitDepth == 12)
             libname = "libx265_main12" ext;
         else if (bitDepth == 10)
-            libname = "libx265_main10" ext;
+            libname = DEB_HOST_MULTIARCH "x265-10bit/libx265" ext;
         else if (bitDepth == 8)
-            libname = "libx265_main" ext;
+            libname = DEB_HOST_MULTIARCH "libx265" ext;
         else
             return NULL;
 
