Index: gst-libav-1.6.1/common/m4/gst-plugindir.m4
===================================================================
--- gst-libav-1.6.1.orig/common/m4/gst-plugindir.m4
+++ gst-libav-1.6.1/common/m4/gst-plugindir.m4
@@ -6,12 +6,12 @@ dnl AC_SUBST plugindir, to be used in Ma
 AC_DEFUN([AG_GST_SET_PLUGINDIR],
 [
   dnl define location of plugin directory
-  AS_AC_EXPAND(PLUGINDIR, ${libdir}/gstreamer-$GST_API_VERSION)
+  AS_AC_EXPAND(PLUGINDIR, ${libdir})
   AC_DEFINE_UNQUOTED(PLUGINDIR, "$PLUGINDIR",
     [directory where plugins are located])
   AC_MSG_NOTICE([Using $PLUGINDIR as the plugin install location])
 
   dnl plugin directory configure-time variable for use in Makefile.am
-  plugindir="\$(libdir)/gstreamer-$GST_API_VERSION"
+  plugindir="\$(libdir)"
   AC_SUBST(plugindir)
 ])
