Index: janus-gateway-audioroom-0.0.1/libjanus_audioroom.c
===================================================================
--- janus-gateway-audioroom-0.0.1.orig/libjanus_audioroom.c
+++ janus-gateway-audioroom-0.0.1/libjanus_audioroom.c
@@ -366,19 +366,19 @@ record_file =	/path/to/recording.wav (wh
  * \ref plugins
  */
 
-#include "plugins/plugin.h"
+#include "janus/plugin.h"
 
 #include <jansson.h>
 #include <opus/opus.h>
 #include <sys/time.h>
 
-#include "debug.h"
-#include "apierror.h"
-#include "config.h"
-#include "mutex.h"
-#include "rtp.h"
-#include "rtcp.h"
-#include "utils.h"
+#include "janus/debug.h"
+#include "janus/apierror.h"
+#include "janus/config.h"
+#include "janus/mutex.h"
+#include "janus/rtp.h"
+#include "janus/rtcp.h"
+#include "janus/utils.h"
 
 
 /* Plugin information */
